src/arrays/programs_involving_multidimensional_data/easy/q8/MainTest.java

```diff
@@ -0,0 +1,69 @@
+package arrays.programs_involving_multidimensional_data.easy.q8;
+
+import global.BaseRandomTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+public class MainTest extends BaseRandomTest {
+    // Parsons
+
+    public static int n = 5;
+    public static int em = 5;
+    public static int om = 3;
+
+    public Clause[] testSentence() {
+        Clause[] c = new Clause[4 * n * n + 2 * n];
+        int t = 0;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                c[t++] = new RandomInteger(1, 1001, i + " " + j);
+                c[t++] = new StringLiteral(" ");
+            }
+            c[t++] = new NewLine();
+        }
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                c[t++] = new IntegerLiteral(i + "x" + j);
+                c[t++] = new StringLiteral(" ");
+            }
+            c[t++] = new NewLine();
+        }
+        return c;
+    }
+
+    public void runMain() {
+        EvenOrOdd.main(new String[0]);
+    }
+
+    public static void answerFor(int[][] arr) {
+        for (int i = 0; i < arr.length; i++) {
+            for (int j = 0; j < arr[0].length; j++) {
+                if (arr[i][j] % 2 == 0)
+                    arr[i][j] *= em;
+                else
+                    arr[i][j] *= om;
+            }
+        }
+    }
+
+    @Test
+    public void printsCorrectOutput() throws InvalidClauseException {
+        int[][] arr = new int[n][n];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                arr[i][j] = Integer.parseInt(getItemByName(i + " " + j));
```

Comment:
consider moving 
```Java
if (arr[i][j] % 2 == 0)
                    arr[i][j] *= em;
                else
                    arr[i][j] *= om;
```
Inside this for loop so that an extra method is not needed